activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -1113,7 +1113,7 @@ def log(sql, name = "SQL", binds = [], type_casted_binds = [], statement_name =
 
         def transform_query(sql)
           ActiveRecord.query_transformers.each do |transformer|
-            sql = transformer.call(sql)
+            sql = transformer.call(sql, self)
```

Comment:
I don't think those transformers were intended to be public API, so I'm not bothering with allowing transformers that accept only one parameter. But let me know if I got it work @byroot about the intention of this being public.